### PR TITLE
Feat - OAuth on Development Mode (Hot Reload ON)

### PR DESCRIPTION
**Pull request by thalesbrant**
Fixing the #761 .

This fix adds one env var on the UI to apply a redirection only on Development mode.
This fix makes by default the next_url to be the PUBLIC_URL informed for proper redirection after callback execution.

### CODE REVIEW

File: querybook/server/app/auth/__init__.py [hunk 1]
```diff
@@ -48,8 +48,11 @@ def get_login_config():
     global auth
     global login_config
     if login_config is None:
+        oauth_url = ""
         has_login = hasattr(auth, "login_user_endpoint")
         has_signup = hasattr(auth, "signup_user_endpoint")
+        has_oauthURL = hasattr(auth, "oauth_authorization_url")
```
Comment (czgu) [inline querybook/server/app/auth/__init__.py:54]: has_oauth_url

File: querybook/webapp/components/UserLoader/UserLoader.tsx [hunk 2]
```diff
@@ -31,8 +32,17 @@ export const UserLoader: React.FunctionComponent = ({ children }) => {
...
-                        data: { has_login: hasLogin, has_signup: hasSignup },
+                        data: {
+                            has_login: hasLogin,
+                            has_signup: hasSignup,
+                            oauth_url: oauthURL,
+                        },
                     } = await UserResource.getLoginMethods();
+
+                    if (getEnvironment() !== 'production' && oauthURL) {
+                        console.log('webpack is running');
```
Comment (czgu) [inline querybook/webapp/components/UserLoader/UserLoader.tsx:43]: remove console.log before merge

**Review by czgu — COMMENTED** [review]
minor style comments, lgtm

File: querybook/server/app/auth/__init__.py (hunk 1 above)
Comment (thalesbrant) [reply querybook/server/app/auth/__init__.py]: ok

File: querybook/webapp/components/UserLoader/UserLoader.tsx (hunk 2 above)
Comment (thalesbrant) [reply querybook/webapp/components/UserLoader/UserLoader.tsx]: ok

**Review by czgu — APPROVED** [review]

**Comment by czgu** [comment]
Thanks for adding oauth for local dev, this troubled me for a long time 😄

**Pull request merged**